packages/contract-wrappers/src/contract_wrappers/erc721_token_wrapper.ts

```diff
@@ -0,0 +1,439 @@
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setProxyApprovalForAllAsync(
+        tokenAddress: string,
+        ownerAddress: string,
+        isApproved: boolean,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        const proxyAddress = this._erc721ProxyWrapper.getContractAddress();
+        const txHash = await this.setApprovalForAllAsync(tokenAddress, ownerAddress, proxyAddress, isApproved, txOpts);
+        return txHash;
+    }
+    /**
+     * Set or reaffirm the approved address for an NFT
+     * The zero address indicates there is no approved address. Throws unless `msg.sender` is the current NFT owner,
+     * or an authorized operator of the current owner.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the Approval event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   approvedAddress The hex encoded user Ethereum address you'd like to set approval for.
+     * @param   tokenId         The identifier for an NFT
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setApprovalAsync(
+        tokenAddress: string,
+        approvedAddress: string,
+        tokenId: BigNumber,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        assert.isETHAddressHex('tokenAddress', tokenAddress);
+        assert.isETHAddressHex('approvedAddress', approvedAddress);
+        assert.isBigNumber('tokenId', tokenId);
+        const normalizedTokenAddress = tokenAddress.toLowerCase();
+        const normalizedApprovedAddress = approvedAddress.toLowerCase();
+
+        const tokenContract = await this._getTokenContractAsync(normalizedTokenAddress);
+        const tokenOwnerAddress = await tokenContract.ownerOf.callAsync(tokenId);
+        const txHash = await tokenContract.approve.sendTransactionAsync(normalizedApprovedAddress, tokenId, {
+            gas: txOpts.gasLimit,
+            gasPrice: txOpts.gasPrice,
+            from: tokenOwnerAddress,
+        });
+        return txHash;
+    }
+    /**
+     * Set or reaffirm 0x proxy as an approved address for an NFT
+     * Throws unless `msg.sender` is the current NFT owner, or an authorized operator of the current owner.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the Approval event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   tokenId         The identifier for an NFT
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setProxyApprovalAsync(
+        tokenAddress: string,
+        tokenId: BigNumber,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        const proxyAddress = this._erc721ProxyWrapper.getContractAddress();
```

Comment:
Missing assertions.

Reply:
Those methods are just convenience methods that call other general methods under the hood. Those have all assertions.